pipelines/controller/src/main/java/org/openmrs/analytics/PipelineManager.java

```diff
@@ -432,9 +493,60 @@ private void createHiveResourceTables(
     }
   }
 
+  /** This method captures the given exception into a file rooted at the dwhRoot location. */
+  void captureError(String dwhRoot, Exception e) {
+    try {
+      if (!Strings.isNullOrEmpty(dwhRoot)) {
+        String stackTrace = ExceptionUtils.getStackTrace(e);
+        DwhFiles.forRoot(dwhRoot)
+            .writeToFile(ERROR_FILE_NAME, stackTrace.getBytes(StandardCharsets.UTF_8));
+      }
+    } catch (IOException ex) {
+      logger.error("Error while capturing error to a file", ex);
+    }
+  }
+
+  /** Sets the details of the last pipeline run with the given dwhRoot as the snapshot location. */
+  void setLastRunDetails(String dwhRoot, Status status) {
+    DwhRunDetails dwhRunDetails = new DwhRunDetails();
+    try {
+      DwhFiles dwhFiles = DwhFiles.forRoot(dwhRoot);
+      String startTime = dwhFiles.readTimestampFile(DwhFiles.TIMESTAMP_FILE_START).toString();
+      dwhRunDetails.setStartTime(startTime);
+      if (status == Status.SUCCESS) {
+        String endTime = dwhFiles.readTimestampFile(DwhFiles.TIMESTAMP_FILE_END).toString();
+        dwhRunDetails.setEndTime(endTime);
+      } else {
+        dwhRoot = dwhRoot.endsWith("/") ? dwhRoot : dwhRoot + "/";
+        dwhRunDetails.setLogFilePath(dwhRoot + ERROR_FILE_NAME);
+      }
+      dwhRunDetails.setStatus(status);
+      dwhRunDetails.setParquetPath(dwhRoot);
+      this.lastRunDetails = dwhRunDetails;
+    } catch (IOException e) {
+      logger.error("Error in reading timestamp files", e);
+      throw new RuntimeException(e);
+    }
+  }
+
   public enum LastRunStatus {
     NOT_RUN,
     SUCCESS,
     FAILURE
   }
+
+  @Data
+  public class DwhRunDetails {
+
+    private String startTime;
+    private String endTime;
+    private Status status;
+    private String parquetPath;
```

Comment:
Are we using this `parquetPath` anywhere?

Reply:
No, its not used. I have removed the same.